2.47

ATSchoolENikonov opened the pull request:
Сделал часть 1, 2, 4,5 задания
Позже докину оставшиеся, но хочу, чтобы уже можно было по ним посмотреть, есть ли косяки

methshest committed:
2.47
Сделал часть 1, 2, 4,5 задания
Позже докину оставшиеся, но хочу, чтобы уже можно было по ним посмотреть, есть ли косяки

Separador commented on Home_2.47/Task_1.java:
```diff
@@ -0,0 +1,13 @@
+        Integer m = Integer.valueOf(args[0]);
+        Integer n = Integer.valueOf(args[1]);
+        int x = m;
+        for (int i = 0; i < (n - m); i++) {
```
С заданием ок. Ты можешь вместо `i` сразу `x` использовать, так будет проще.

Separador commented on Home_2.47/Task_2.java:
```diff
@@ -0,0 +1,19 @@
+        int x = 0;
+        if (m > 0 && n >= 0) {
+            for (int i = 1; i < m; i++) {
+                x += 10;
```
По условию каждая следующая косилка работает, а каждая следующая на 10 минут больше, чем предыдущая. Если предыдущая работала 1 час, то следующая буде работать 1 час + 10 минут. А за ней идущая - 1 час 20 минут и т.д.

Separador commented on Home_2.47/Task_4.java:
```diff
@@ -0,0 +1,36 @@
+    boolean isPrime(int x) {
+        boolean check = true;
+        if (x < 2) {
+            check = false;
```
По заданию - ок. Тут ты можешь сразу делать `return false`, тк уже с числом все понятно, дальшие вычисления не нужны.

Separador commented on Home_2.47/Task_4.java:
```diff
@@ -0,0 +1,36 @@
+        }
+        for (int i = 2; i < x; i++) {
+            if (x % i == 0) {
+                check = false;
```
Тоже тут можешь сразу делать `return false`. Например, если x = 1000000, то ты уже на первой итерации будешь знать, что число составное. Но все равно продолжишь ходить по циклу до конца.

Separador reviewed (changes requested):
Немного ошибся во второй + есть несколько необязательных улучшений

methshest committed:
2.47
Доделал часть 1, 2, 4,5 задания
Сделал 6 задание, вроде выводит таймер посекундно

Separador reviewed (changes requested):
Шестое норм, необязательные улучшения:
- обрати внимание на пример вывода в задании
00:01
00:00
по завершению печатается 00:00.
- обрати внимание на формат вывода. посмотри  метод System.out.printf и возможности форматирования вывода чисел https://docs.oracle.com/javase/7/docs/api/java/util/Formatter.html в разделе Numeric

methshest committed:
2.47
накинул вариант 3 задачи(знаю, что неправильно, но вообще не могу сообразить, как она делается)+ доделал формат вывода для таймера

Separador reviewed (changes requested):
Неплохо.
На аргументе  `5 4` печатает
```
1 2 3 4 
8 
```
а ожидается
```
1 2 3 4 
      5 
```

На аргументе `10 3` получается
```
1 2 3 
6 5 4 
7 8 9 
12 
```
вместо
```
1 2 3
6 5 4
7 8 9
    10
```

На аргументе

methshest committed:
2.47
Исправление задачи
